### PR TITLE
Port to Remark 4

### LINKED ISSUE #15: Is this compatible with remark 4?
I just tried to update a boilerplate of mine to use the newly released remark 4. For some reason it didn't process correctly there (output remained as markdown). Is remark-react compatible with it?

### PR DESCRIPTION
**Pull request by bebraw**
Adapted based on https://github.com/wooorm/remark-vdom/commit/ad20cf9f71c258aca23aad277715ad2fe82bb0ff and tested against a local project. Renders just fine there (images etc.). Obviously I might have missed something.

Closes #15.

### CODE REVIEW

**Comment by bebraw** [comment]
Those tests need some serious work. Would it be acceptable to drop `0.13` support? That's kind of old anyway.

File: lib/compilers.js [hunk 1]
```diff
@@ -342,7 +343,7 @@ function footnote(node) {
         'position': node.position
     });
 
-    return self.footnoteReference({
+    return self.visitors.footnoteReference({
```
Comment (wooorm) [inline lib/compilers.js]: Your missing the context object for `footnoteReference` here, it should be invoked with the compiler (`self`), e.g. `...footnoteReference.call(this, ...`!

Comment (bebraw) [inline lib/compilers.js]: Fixed. Thanks.

**Comment by tmcw** [comment]
Yep, I'm fine with dropping v0.13 support.

**Comment by bebraw** [comment]
@tmcw Cool. I did that and made sure tests pass. Just needed to update deps...

**Comment by thangngoc89** [comment]
Cool. Tracking this. https://github.com/MoOx/statinamic/issues/149

**Comment by tmcw** [comment]
:+1: looks great, merging.

**Comment by tmcw** [comment]
Released as 2.0.0

**Comment by thangngoc89** [comment]
:+1:

**Comment by AlastairTaft** [comment]
:+1:

**Comment by wooorm** [comment]
Awesome! Thanks @bebraw and @tmcw!

**Pull request merged**